### ISSUE TITLE
JavaFX's Application class cannot find the main class, despite being invoked from the former.

### ISSUE THREAD

**Issue by Xemiru** [libgdx/packr#67]
The executable shuts off the JVM as soon as `Application.launch(args)` from JavaFX is ran. Test message was placed directly after the invocation and was not displayed on the console.

``` java
    public static void main(String[] args) {
        System.out.print("test0");
        initRuntime();

        String devCheck = System.getenv("tailor-dev-mode");
        if (devCheck != null && devCheck.equalsIgnoreCase("true")) {
            LaunchOptions options = new LaunchOptions(true);

            log.info("Running in development mode!");
            launchGame(options);
            return;
        }

        System.out.println("test1");
        Application.launch(args);
        System.out.println("test2"); // never shown
    }
```

```
Loading JVM runtime library ...
Passing VM options ...
  # -Xms512M
  # -Xmx1536M
Creating Java VM ...
Passing command line arguments ...
Loading JAR file ...
Invoking static me.scarlet.undertailor.desktop.DesktopLauncher.main() function ...
test0test1
Destroyed Java VM ...
Press ENTER key to exit.
```

Built using a 64bit Oracle JRE, unminified for the purpose of testing.

**Comment by Xemiru** [comment]
https://github.com/libgdx/packr/blob/master/src/main/native/packr/src/packr.cpp#L561

I don't know C++, but I'd assume this is the faulting line, as it terminates when the main method closes and not when all non-daemon threads are dead.

**Comment by code-disaster** [comment]
I never used JavaFX, but the Application class does a lot of reflection work. Maybe try to try {} catch {} everything and see if there's an unhandled exception slipping through.

**Comment by Xemiru** [comment]
It's throwing a `ClassNotFoundException` over the `DesktopLauncher` class.

Which doesn't make sense, since the `main` method lives inside that class; it shouldn't even need to ask for it. The program otherwise works perfectly fine outside of packr's executable, making it to the game's launcher window.

```
Invoking static me.scarlet.undertailor.desktop.DesktopLauncher.main() function ...
test0test1
java.lang.RuntimeException: java.lang.ClassNotFoundException: me.scarlet.undertailor.desktop.DesktopLauncher
        at javafx.application.Application.launch(Application.java:260)
        at me.scarlet.undertailor.DesktopLauncher.main(DesktopLauncher.java:78)
Caused by: java.lang.ClassNotFoundException: me.scarlet.undertailor.desktop.DesktopLauncher
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at javafx.application.Application.launch(Application.java:248)
        ... 1 more
test 2
Destroyed Java VM ...
Press ENTER key to exit.
```

**Comment by Xemiru** [comment]
@code-disaster It's still very odd that this occurs, luckily the `launch` method has alternate parameters where I can specify the DesktopLauncher class. I'll leave this issue open as this seems to be local to `packr` only.

```java
// fix
Application.launch(DesktopLauncher.class, args);
```

**Comment by shaigem** [comment]
The same issue still occurs even with the fix in version 2.0.

```java
public static void main(String[] args) {
        System.out.println("hello world 1");
        try {
            Application.launch(JavaFXApp.class, args);
            System.out.println("hello world 2");

        } catch (Exception ex) {
            ex.printStackTrace();
        }
        System.out.println("hello world 3");
    }

Only hello world 1 gets outputted and the application just exits.

**Comment by guikeller** [comment]
I have too faced the same issue as mentioned by the others when distributing an app that uses Java FX.

To get around this issue, I created a bash script that unzips the JDK and puts the JRE in a specific folder.
As a workaround to start the app I created a: .command, .bat, .sh - it is not pretty but _it works_.

@code-disaster - A suggestion would be to have a **minimizeJre='skip'** and distribute the JRE in full.
